core/src/main/java/jenkins/widgets/HistoryPageFilter.java

```diff
@@ -101,85 +105,100 @@ public void setSearchString(@Nonnull String searchString) {
     /**
      * Add build items to the History page.
      *
-     * @param items The items to be added. Assumes the list of items are in descending queue ID order i.e. newest first.
+     * @param runItems The items to be added. Assumes the list of items are in descending queue ID order i.e. newest first.
      */
-    public void add(@Nonnull List<T> items) {
-        if (items.isEmpty()) {
+    public void add(@Nonnull Iterable<T> runItems) {
+        addInternal(runItems);
+    }
+
+    /**
+     * Add run items and queued items to the History page.
+     *
+     * @param runItems The items to be added. Assumes the list of items are in descending queue ID order i.e. newest first.
+     * @param queueItems The queue items to be added.
+     */
+    public void add(@Nonnull Iterable<T> runItems, @Nonnull List<Queue.Item> queueItems) {
```

Comment:
Missing `@since` for public API. Missing `@Restricted` if not supposed to be used from plugins (unclear).